redact_sample.py

```diff
@@ -0,0 +1,117 @@
+import argparse
+import csv
+import logging
+import os
+import sys
+from typing import Iterable
+
+from generate_sample_file import SampleGenerator
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
+SampleGen = SampleGenerator()
+SampleGen.ESTAB_TYPES = ['HALL OF RESIDENCE', 'CARE HOME', 'HOSPITAL', 'HOSPICE', 'MENTAL HEALTH HOSPITAL',
+                         'MEDICAL CARE OTHER', 'BOARDING SCHOOL', 'HOTEL',
+                         'YOUTH HOSTEL', 'HOSTEL', 'EDUCATION OTHER', 'PRISON', 'STAFF ACCOMMODATION', 'CAMPHILL',
+                         'HOLIDAY PARK', 'HOUSEHOLD', 'RESIDENTIAL CARAVAN', 'RESIDENTIAL BOAT', 'GATED APARTMENTS',
+                         'FOREIGN OFFICES', 'CASTLES', 'EMBASSY', 'CARAVAN', 'MARINA']
+SampleGen.read_words()
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description='Load a sample file into response management.')
+    parser.add_argument('sample_file_path', help='path to the sample file', type=str)
+    return parser.parse_args()
+
+
+def redact_sample_file(sample_file_path, output_file_path):
+    with open(sample_file_path) as sample_file:
+        redact_sample(sample_file, output_file_path)
+    sample_file.close()
+
+
+def redact_sample(sample_file: Iterable[str], output_file_path):
```

Comment:
I'm not a pythonista, but it seems strange to mix type checking with untyped parameters. For better code quality, wouldn't it be better to use type checking throughout, consistently?

Reply:
Your suggestion will make reading the code easier, going to make these changes.

Reply:
Just to be clear, those are type hints not enforced type checking, even with this line of code if you passed something that was not an Interable[str] to this function as an object sample_file, python wouldn't complain at execution ( unless the object caused a specific error ) - However I do think that the type hints make the code more readable and thus have made the changes. Additionally, all of these variables are generated within the script and so unless further changes are made they will always be the type required.